code/libraries/joomlatools/library/http/client/client.php

```
@@ -0,0 +1,277 @@
+ */
+
+/**
+ * Http Client
+ *
+ * @author  Johan Janssens <https://github.com/johanjanssens>
+ * @package Koowa\Library\Http\Client
+ */
+class KHttpClient extends KObject implements KHttpClientInterface
+{
+    /**
+     * Initializes the options for the object
+     *
+     * Called from {@link __construct()} as a first step of object instantiation
+     *
+     * @param   KObjectConfig $config  An optional KObjectConfig object with configuration options
+     * @return  void
+     */
+    protected function _initialize(KObjectConfig $config)
+    {
+        $config->append(array(
+            'user_agent'      => 'Joomlatools/Framework/'. Koowa::getInstance()->getVersion(),
+            'follow_location' => 0,
+        ));
+
+        parent::_initialize($config);
+    }
+
+    /**
+     * Send a http request
+     *
+     * @param  KHttpRequestInterface $request   The http request object
+     * @throws RuntimeException If the request failed
+     * @return  KHttpResponseInterface
+     */
+    public function send(KHttpRequestInterface $request)
+    {
+        if(!ini_get('allow_url_fopen')) {
+            throw new RuntimeException('Cannot use a stream transport when "allow_url_fopen" is disabled.');
+        }
+
+        $headers = $request->getHeaders()
+            ->set('Connection', 'close');
+
+        $context = stream_context_create(array('http' => array(
+            'user_agent'       => $this->getConfig()->user_agent,
+            'protocol_version' => $request->getVersion(),
+            'header'           => (string) $headers,
+            'follow_location ' => $this->getConfig()->follow_location,
+            'method'           => $request->getMethod() ?? 'GET',
+            'content'          => (string) $request->getContent(),
+        )));
+
+        $url     = $request->getUrl();
+        $content = file_get_contents($url, false, $context);
```

@johanjanssens Shouldn't we suppress errors here? Since we are already handling it down in the next block

---

Done and further improved the handling